test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+ * Each search_near call with prefix enabled is verified using the default search_near.
+ */
+class search_near_02 : public test_harness::test {
+    public:
+    search_near_02(const test_harness::test_args &args) : test(args) {}
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *,
+      test_harness::configuration *config, test_harness::workload_tracking *) override final
+    {
+        /*
+         * The populate phase only creates empty collections. The number of collections is defined
+         * in the configuration.
+         */
+        int64_t collection_count = config->get_int(COLLECTION_COUNT);
+
+        logger::log_msg(
+          LOG_INFO, "Populate: " + std::to_string(collection_count) + " creating collections.");
+
+        for (int64_t i = 0; i < collection_count; ++i)
+            database.add_collection();
+
+        logger::log_msg(LOG_INFO, "Populate: finished.");
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        /* Each insert operation will insert new keys in the collections. */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        /* Helper struct which stores a pointer to a collection and a cursor associated with it. */
+        struct collection_cursor {
+            collection_cursor(collection &coll, scoped_cursor &&cursor)
+                : coll(coll), cursor(std::move(cursor))
+            {
+            }
+            collection &coll;
+            scoped_cursor cursor;
+        };
+
+        /* Collection cursor vector. */
+        std::vector<collection_cursor> ccv;
+        uint64_t collection_count = tc->db.get_collection_count();
+        uint64_t collections_per_thread = collection_count / tc->thread_count;
+
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tc->thread_count == 0);
+        for (int i = tc->id * collections_per_thread;
+             i < (tc->id * collections_per_thread) + collections_per_thread && tc->running(); ++i) {
+            collection &coll = tc->db.get_collection(i);
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
+            ccv.push_back({coll, std::move(cursor)});
+        }
+
+        std::string key;
+        uint64_t counter = 0;
+
+        while (tc->running()) {
+
+            auto &cc = ccv[counter];
+            tc->transaction.begin();
+
+            while (tc->transaction.active() && tc->running()) {
+
+                /* Generate a random key. */
+                key = random_generator::instance().generate_random_string(tc->key_size);
+
+                /* Insert a key value pair. */
+                if (tc->insert(cc.cursor, cc.coll.id, key)) {
+                    if (tc->transaction.can_commit())
```

Comment:
If this returns false, we also need to rollback.

Reply:
I disagree, `can_commit` here checks if we have performed enough operations in the current transaction before being able to commit.

Reply:
oh I see, I was thinking of the scenario of where can_commit returns false, and we begin another transaction (which will assert). In that case, i think `tc->transaction.try_begin()` might be a better option here.

Reply:
mb i didn't read the code correctly.